### ISSUE TITLE
How about reading  from memory?

### ISSUE THREAD

**Issue by msnh2012** [sreiter/stl_reader#11]
Nice work bro. How about reading from memory?

**Comment by sreiter** [comment]
Thanks for the nice words :). With a rather small refactoring, this should very well be possible, e.g., by providing an overload of ReadStlFile_... which takes an `std::istream`. The current file based overloads could then simply handle file loading and forward their streams to those new overloads.

However, personally I didn't encounter the use case yet... would you mind sharing yours?

**Comment by msnh2012** [comment]
Yep, here is my case. I want to encode the STL file with a key, which needs to be read from memory when decoding.
